content/blog/2018-11-11-colocr.md

```diff
@@ -0,0 +1,106 @@
+  roi_select(threshold = 90, n = 3) %>%
+  roi_test()
+```
+
+And the results can be visualized in a graph.
+
+![](/img/blog-images/2018-11-11-colocr/graph.png)
+
+
+The following code is to generate the graph.
+
+```r
+## load required libraries
+library(tidyverse)
+
+# make the graph of calculated correlations
+names(tst) <- str_split(fls, pattern = '/', simplify = TRUE)[, 1]
+
+bind_rows(tst, .id = 'protein') %>%
+  group_by(protein) %>%
+  mutate(ave = mean(pcc),
+         sd = sd(pcc),
+         upper = ave + sd,
+         lower = ave - sd) %>%
+  ggplot(aes(x = protein, y = pcc)) +
+  geom_jitter() +
+  geom_point(aes(y = ave), color = 'red') +
+  geom_linerange(aes(ymax = upper, ymin = lower), color = 'red') +
+  lims(y = c(0, 1.1)) +
+  labs(x = '', y = "Pearson's Correlation Coefficient (PCC)") +
+  theme_bw()
+```
+
+While developing `colocr`, I focused on breaking down the steps of the analysis into intuitive pieces. Hopefully, in a way that the potential user, or "myself in a few months", can understand easily. By keeping the number of the functions limited, the inputs and the outputs consistent and making things visual as much as possible, I hope using the package needs less investment by the user. I hope biologist who are not familiar with [R](https://cran.r-project.org) can still find this work useful. One thing helped me organize and iterate on the code. This was developing a shiny app as an interface to the package.
+
+## Developing a shiny app in an R package
+
+At first, I thought of adding a shiny app as an extra feature for users who might need it. I quickly realized that for this use case, selecting regions of interest might require trial-and-error on the side of the user to arrive at suitable parameters for the selection. Having to develop the app forced me to think about the organizing the functions, their input and output in a certain way. Namely, mapping the functions to the steps of the analysis that the user has to think about and making visual output a priority. Finally, testing the app exposed errors in the intermediates which might have been buried in an otherwise seemingly fine [R](https://cran.r-project.org) objects. Overall, I learned a thing or two about writing shiny apps to go with [R](https://cran.r-project.org) packages when possible even if it won't make the cut in the release.
+
+* I found it useful to work on the app as early as possible and improving it side by side with the core functionality. As explained earlier, this helped me to think about the inputs and the output from a user's perspective.
+*  I learnt to isolate the source code for the app from that of the package functions as much as possible. One may host the app code in a separate repository and including it in the project as a **git submodule**. Some might find this necessary, but I found it easier this way to keep tracking of the changes in the app and testing it on Travis.
+
+## Submitting R packages for reviewing on ROpenSci
```

Comment:
ROpenSci -> rOpenSci 
Please use this capitalization throughout post. No need to include URL since it's on our site